src/Filters/Filter.php

```diff
@@ -179,7 +179,7 @@ protected function shouldProcessFile($path)
         // complete extension list and make sure one is allowed.
         $extensions = [];
         array_shift($fileParts);
-        foreach ($fileParts as $part) {
+        while (empty($fileParts) === false) {
```

Comment:
First I was a bit confused how this loop works, but now with the while loop it is much more clear that the fileParts array is being modified.

Reply:
This code is not covered by tests.

Reply:
should I add test coverage for this as part of this PR?

Reply:
Not necessarily, but it does mean it needs to be (manually) tested carefully, as I've seen too many `while` loops which never end.

Of course, if you're willing to add tests for that class, I won't say "no", but might be better to do that in a separate PR ?

Reply:
ok, I think the while loop is quite short and easy to understand, so I'm not worried. Will leave this in the PR.